pkg/secretless/config/v2/config_test.go

```diff
@@ -0,0 +1,94 @@
+package v2
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+const sampleConfigStr = `
+version: 2
+services:
+  postgres-db:
+    protocol: pg
+    listenOn: tcp://0.0.0.0:5432 # can be a socket as well (same name for both)
+    credentials:
+      address: postgres.my-service.internal:5432
+      password:
+        from: vault
+        get: name-in-vault
+      username:
+        from: env
+        get: USERNAME
+    config:  # this section usually blank
+      optionalStuff: blah
+`
+
+func sampleConfig() (*Config, error) {
+	configFileContents := []byte(sampleConfigStr)
+	return NewConfig(configFileContents)
+}
+
+func TestNewConfig(t *testing.T) {
```

Comment:
Probably not a blocker but tests in this file should be split out to `config_http_test`, `credential_test`, `config_yaml`, and `service_v1` tests. We also do a pretty slim job imo on testing here but it's probably enough for the first pass.